Lib/zipfile.py

```diff
@@ -1112,6 +1115,19 @@ def seek(self, offset, whence=0):
             if self._decrypter is not None:
                 self._init_decrypter()
 
+        # Fast seek uncompressed unencrypted file
+        if self._compress_type == ZIP_STORED and self._decrypter == None and read_offset > 0:
+            # disable CRC checking after first seeking - it would be invalid
```

Comment:
This isn't going to be desirable for all users in all cases. It is silently disabling CRC checking for a read performance optimization. What if the user does not want this to happen and wants the CRC check?

As annoying as accumulating feature flags can be, I think this warrants a bool parameter on the `class ZipFile:` constructor. A new `fast_stored_seek_nocrc=False` argument perhaps. That does require people to opt-in to this performance gain, but making the choice to not have what came for free in the past is better than pulling the rug out from underneath existing users who might've been relying on it.

Reply:
Sure - I can do this.

Anyway I don't think it's neccessary - crc error is only triggered when whole file is read anyway.
If you are not using seek and you are reading whole file to the end - it would raise an exception (after my patch and before too).
If you are seeking on the file you'll probably not read the whole file anyway and it won't raise.

Reply:
Done as opt-out, if you wish I can change it to opt-in.

I'd honestly prefer to have it as opt-out, as most people using seek don't have possiblity to use CRC anyway as their code reads only small chunks of the files (that's what seek is for, right?).
Making it opt-out would make it work with legacy, unmaintained code and libraries which is my actual use-case.
If someone really wants to - they can force old mechanism now, but by default fast one would be used.

Reply:
Oh and I forgot to tag you: @gpshead

Reply:
I do not like this options. It complicates the interface and only confuses users. In most cases it does not have any effect. And I agree with @JuniorJPDJ that it is unlikely that you use `seek()`, read the uncompressed file to the end, and need a CRC check. Note that CRC does not guarantee that the archive was not corrupted. If you have doubts about integrity of your data, better to use more reliable methods.

Reply:
Thanks. Rethinking. @serhiy-storchaka is right here. Given the required incantations to trigger this optimization and circumstances it'll typically be used in... Lets go back to removing the "fast_stored_seek_nocrc" option that I asked for.  sorry for the churn  :)